### ISSUE TITLE
Define available hours in Gantt chart to allow proper usage of hour-tasks

### ISSUE THREAD

**Issue by maxschons** [mermaid-js/mermaid#1555]
**Is your feature request related to a problem? Please describe.**
I want to use the Gantt chart for both tasks that will stretch over multiple days and some that only last for hours. Currently days last 24h which leads to 6 4h tasks in one day when they follow each other sequentally. It seems reasonable to only have 8 available hours per day for task planning (or something in this area). So in summary working with tasks that last hours will create wrong estimates for when I will finish my tasks

**Describe the solution you'd like**
I would like to be able to set my available hours per day e.g. 2, 4, 8 or 12 (ideally for each day) and define how many hours a 1day task has. The benefit here is that it would scale well with smaller projects that only have been allocated x hours. I could see for instance, that a 1d task (defined with 1d = 8h) would stretch over 2 days in a 4hour work day chart. 

**Describe alternatives you've considered**
Alternatives include allowing me to write "0.5d" or "0.25d" instead. I could think of my 4 hour task as a 0.5 day task and Mermaid in the background generates a 12h block. Similarly for 2h and 0.25 days => 6h block. This would probably be an easier fix but will lack customizability.

**Additional context**
Professional Gantt chart software often has these capabilities and it creates a much better planning experience.

**Comment by RobSeder** [comment]
## Example

Here's another example:

```mermaid
gantt
    title Project Planning by Person-Hours
    section Research
	    Vendor Research       :vr, 2014-01-01 08:00, 20h
	    Onboarding              :co, after vr, 15h
    section Consultant Work
	    Analysis              :a1, after co, 80h
	    Documentation         :after a1, 60h
```

with something like this for code:

```text
gantt
    title Project Planning by Person-Hours
    section Research
	    Vendor Research       :vr, 2014-01-01 08:00, 20h
	    Onboarding            :co, after vr, 15h
    section Consultant Work
	    Analysis              :a1, after co, 80h
	    Documentation         :after a1, 60h
```

So, the idea is that you'd want to say "this item will take 20 hours" which means 20 WORKING hours, meaning 8 hours a day typically. Or for people on a 4 Day Workweek, a 10-hour work day. So, being able to set working hours for the `exclude` property would be useful for sure.

However, how it currently works is this seems to count full, 24-hour days, as if the worker is working 24-hours per day.

---

## Potential Solution

What comes to mind for me, which would likely be the easiest to implement, but not the most elegant - is to just allow setting working hours by day of week. For example:

```text
gantt
    title Project Planning by Person-Hours
    weekstartday Sunday
    availablehours 0,8,8,8,8,8,0
    ...
```

**Comment by thiagoolsilva** [comment]
@RobSeder any update for your solution? I'd love to solve this problem but I don't have any success to solve the related problem with my Gantt diagram.

**Comment by RobSeder** [comment]
@thiagoolsilva no, sorry. This is going to have to be something that is supported in the library, itself. I don't have any workarounds either.

**Comment by thiagoolsilva** [comment]
Thanks

**Comment by jgreywolf** [comment]
This is approved, but we need someone to take it on :)

**Comment by robinfehr** [comment]
that would be very useful indeed! i just found out that isn't working after I've put in quite some time into a bigger chart - wrongfully assuming thus far that this for sure must be a default feature.

**Comment by kraxner** [comment]
This seems to be related to #4060, and in this case will be resolved as well by #5403
